lib/validator/validator.go

```diff
@@ -66,6 +89,23 @@ func init() {
 	registerStructValidator(validateNodeSpec, api.NodeSpec{})
 }
 
+// reason returns the provided error reason prefixed with an identifier that
+// allows the string to be used as the field tag in the validator and then
+// re-extracted as the reason when the validator returns a field error.
+func reason(r string) string {
+	return reasonString + r
+}
+
+// extractReason extracts the error reason from the field tag in a validator
+// field error.
+func extractReason(tag string) string {
+	if strings.HasPrefix(tag, reasonString) {
+		return strings.TrimPrefix(tag, reasonString)
+	} else {
+		return ""
```

Comment:
would there ever be a case where `reasonString` is not prefixed? if not we should be returning an error since returning an empty string is a silent failure